src/gui/socketapi.cpp

```diff
@@ -757,11 +757,11 @@ void SocketApi::command_GET_MENU_ITEMS(const QString &argument, OCC::SocketListe
     }
 
     // Virtual file download action
-    if (folder) {
+    if (folder && folder->useVirtualFiles()) {
```

Comment:
Hm, we might want to rename/document this better. Actually, this setting just controls whether new files will start out as virtual or not. It's perfectly possible to switch this of (so all new files are no longer virtual) but still have heaps of virtual files in the sync directories.

Short: I'd show this option even for folders that don't have the option enabled.

Reply:
Right, in the future, we could decouple the two.
The problem is that I don't really want to recursively go into the folder to see if there are placeholders.
(altough maybe a SELECT WHERE type=4 AND IS_PREFIX_PATH_OF might be simple enough)

Reply:
What about moving the check down to the files loop then and only removing the action from directories when the folder setting is disabled? Then people could at least still download files individually if the option is off.